src/webloader.js

```diff
@@ -0,0 +1,126 @@
+import { generate } from './guids.js';
+import EvolvAssetManager from './index.js';
+import EvolvClient from "@evolv/javascript-sdk";
+
+
+function ensureId(evolv, key, session) {
+	let id;
+	if (evolv.context && key in evolv.context) {
+		id = evolv[key];
+	} else {
+		id = evolv.retrieve(key, session);
+		if (!id) {
+			id = generate();
+			evolv.store(key, id, session);
+		}
+	}
+	return id;
+}
+
+function currentScript() {
+	if (document.currentScript) {
+		return document.currentScript;
+	}
+
+	for (let i = 0; i < document.scripts.length; i++) {
+		const script = document.scripts[i];
+		if (script && script.dataset && 'env' in script.dataset) {
+			return script;
+		}
+	}
+}
+
+function injectScript(endpoint, env, uid) {
+	const script = document.createElement('script');
+	script.type = 'text/javascript';
+	script.src = endpoint + '/' + env + '/' + uid + '/assets.js';
```

Comment:
See comment below about endpoint URL.
```suggestion
	script.src = endpoint + '/v1/' + env + '/' + uid + '/assets.js';
```

Reply:
Also I'm familiar with the timing assumptions that this library makes, but you may want to mark this script `defer`. Without `defer`, this script will block the page parsing until the script has been retrieved and processed. It may also execute the script before the DOM has been parsed therefore nullifying its effect. Marking the script with `defer` will not block parsing and will indicate that the script should run only after the page has been parsed. 
```suggestion
	script.src = endpoint + '/v1/' + env + '/' + uid + '/assets.js';
	script.defer = true;
```